src/sysinfo/nvapi_adapter.cpp

```diff
@@ -147,7 +157,51 @@ namespace dxvk {
         return 0;
     }
 
-    bool NvapiAdapter::isVkDeviceExtensionSupported(const std::string name) { // NOLINT(performance-unnecessary-value-param)
+    bool NvapiAdapter::GetLUID(LUID *luid) const {
```

Comment:
I tried to avoid using pointers when not needed internally, so if possible I would prefer passing `luid` by reference here. Alternatively we could split this method into `HasValidLuid()` and `GetLuid()` and then just return the LUID in `GetLuid()`. But not sure if this would just be boilerplate code. What do you think?

Reply:
Does that data have to be allocated or the pointer can just be set to internally-stored data? IIRC, the latter is possible, because most of the pointers are valid only when the nvapi is initialised, and after deinitialisation, the pointers are said to be able to be different. I don't object to it being copied though, I did that myself in my implementation of it.

Reply:
I'd prefer to not split this into `HasValidLuid()` and `GetLuid()` as it just seems a bit too piecemeal for my taste (then clients of the adapter need to actively care about the specifics of LUID validity).

To me the tradeoff for code complexity vs performance benefit sways towards just doing the copy. As far as I'm aware `NvAPI_GPU_GetAdapterIdFromPhysicalGpu()` isn't expected to be in the hot path for any applications, and maybe is called a handful of times.